mne/utils/mixin.py

```diff
@@ -72,10 +73,16 @@ def __hash__(self):
             raise RuntimeError(f"Hashing unknown object type: {type(self)}")
 
 
+SelfEpochs = TypeVar("SelfEpochs", bound="GetEpochsMixin")
```

Comment:
Maybe for forward compatibility you might already want to name it `Self`,  which would minimize the transition.

Reply:
I'm not sure, because of the `bound` param here ... I didn't want to use a too generic name but i'm a little clueless. I can change it if you want!

Reply:
It's up to you. I think it would be neat if you already named it `Self`, so we don't have to change much in the future.

Another issue that comes to mind is that `GetEpochsMixin` is also used for `EpochsSpectrum`. Do you get the correct type annotation for that?

Reply:
I think it was indeed possible to simplify things!! See 
a56a23f

Reply:
Oh wow! No need to set the `bound` class?

Reply:
> Another issue that comes to mind is that `GetEpochsMixin` is also used for `EpochsSpectrum`. Do you get the correct type annotation for that?

I just checked, yes, it works correctly!

Reply:
> Oh wow! No need to set the `bound` class?

Seems so! Pretty cool.